lib/runtime/wasmer/instance.go

```diff
@@ -311,10 +312,10 @@ func (in *LegacyInstance) NetworkService() runtime.BasicNetwork {
 
 // int64ToPointerAndSize converts an int64 into a int32 pointer and a int32 length
 func int64ToPointerAndSize(in int64) (ptr int32, length int32) {
-	return int32(in), int32(in >> 32)
+	return int32(in & 0xffffffff), int32(in >> 32)
 }
 
 // pointerAndSizeToInt64 converts int32 pointer and size to a int64
 func pointerAndSizeToInt64(ptr, size int32) int64 {
-	return int64(ptr) + (int64(size) << 32)
+	return int64(ptr) | (int64(size) << 32)
```

Comment:
also here

Reply:
it should be the same so just curious

Reply:
Bitwise operation is more optimal than arithmetic operation. The result would be the same.